### PR TITLE
SWC-5647 - Entity Badge Icons

### PR DESCRIPTION
**Pull request by nickgros**
Small refinements to prepare for use in SWC

### CODE REVIEW

File: src/lib/containers/EntityBadgeIcons.tsx [hunk 1]
```diff
@@ -305,19 +314,33 @@ export const EntityBadgeIcons = (props: EntityBadgeIconsProps) => {
...
+                  confirmButtonText="Unlink"
+                  confirmButtonVariant="danger"
+                  onConfirm={() => {
+                    unlinkEntity(entityId)
+                    setShowUnlinkConfirmModal(false)
+                  }}
+                  onCancel={() => {
+                    setShowUnlinkConfirmModal(false)
+                  }}
+                />
```
Comment (nickgros) [inline src/lib/containers/EntityBadgeIcons.tsx:333]: Show a warning modal before deleting a link

File: src/lib/utils/hooks/SynapseAPI/entity/queryKeys.ts [hunk 2]
```diff
@@ -101,5 +101,9 @@ export function invalidateAllQueriesForEntity(
   queryClient: QueryClient,
   id: string,
 ) {
+  // TODO: Figure out how to handle cases where the change can affect other entities.
+  // Some examples:
+  //  - deleting an entity should invalidate the parent's children
+  //  - moving an entity should invalidate the old and new parent's children
```
Comment (nickgros) [inline src/lib/utils/hooks/SynapseAPI/entity/queryKeys.ts:107]: Just leaving a note for the future as we begin to do more edit/delete through react-query

File: src/lib/containers/EntityBadgeIcons.tsx [hunk 3]
```diff
@@ -122,18 +122,20 @@ export const EntityBadgeIcons = (props: EntityBadgeIconsProps) => {
...
-  const { data: schemaValidationResults } = useGetValidationResults(entityId, {
+  const { data: boundSchema } = useGetSchemaBinding(entityId, {
     enabled: isInExperimentalMode && inView,
     staleTime: 60 * 1000, // 60 seconds
   })
-  const { data: boundSchema } = useGetSchemaBinding(entityId, {
-    enabled: isInExperimentalMode && inView,
+
+  const { data: schemaValidationResults } = useGetValidationResults(entityId, {
+    enabled: isInExperimentalMode && inView && !!boundSchema,
```
Comment (nickgros) [inline src/lib/containers/EntityBadgeIcons.tsx:138]: Don't fetch validation results unless there is a bound schema

File: src/__tests__/lib/containers/EntityBadgeIcons.test.tsx [hunk 4]
```diff
@@ -312,13 +320,26 @@ describe('EntityBadgeIcons tests', () => {
...
 
+      // Confirmation modal appears
+      await screen.findByText('Confirm Unlink')
+      userEvent.click(await screen.findByRole('button', { name: 'Unlink' }))
+
       await waitFor(() => expect(defaultProps.onUnlinkError).toBeCalled())
     })
   })
 
   describe('displays schema validity', () => {
```
Comment (nickgros) [inline src/__tests__/lib/containers/EntityBadgeIcons.test.tsx:331]: For the tests below
* Ensure MSW reports that a schema is bound, if checking schema validation results
* Adjust what conditions we wait for to ensure test stability

File: src/__tests__/lib/containers/EntityBadgeIcons.test.tsx [hunk 5]
```diff
@@ -331,78 +352,154 @@ describe('EntityBadgeIcons tests', () => {
...
+      await waitFor(() => {
+        icon = screen.getByTestId('annotations-icon')
+        expect(icon.classList.contains('Missing')).toBe(true)
+      })
+
+      userEvent.hover(icon!)
+      await screen.findByText('Missing Annotations')
+    })
+
+    it('Does not fetch validation results if there is no bound schema', () => {
```
Comment (nickgros) [inline src/__tests__/lib/containers/EntityBadgeIcons.test.tsx:464]: New test

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**